service/worker/batcher/workflow.go

```diff
@@ -422,10 +441,11 @@ func processSignalTask(ctx context.Context, limiter *rate.Limiter, task taskDeta
 			return err
 		}
 	}
+	activity.RecordHeartbeat(ctx, task.hbd)
 	return nil
 }
 
-func processTerminateTask(ctx context.Context, limiter *rate.Limiter, task taskDetail, batchParams BatchParams) error {
+func processTerminateOrCancelTask(ctx context.Context, limiter *rate.Limiter, task taskDetail, batchParams BatchParams) error {
```

Comment:
Can we change this to something like the below.. So that we separate out the child logic from the actual task which is a just a simple RPC call.. 

```
func processChildRecursiveTask(ctx context.Context,... , terminateChild bool, task func() error)
```

and call this with 
```
processChildRecursiveTask(ctx, ..., terminateChildren, func() error {
return batcher.svcClient.TerminateWorkflowExecution(newCtx, &shared.TerminateWorkflowExecutionRequest{
			WorkflowExecution: &wf,					Domain:            common.StringPtr(batchParams.DomainName),
			Reason:            common.StringPtr(batchParams.Reason),					WorkflowExecution: &wf,
			Identity:          common.StringPtr(batchWFTypeName),					Reason:            common.StringPtr(batchParams.Reason),
		})					Identity:          common.StringPtr(batchWFTypeName),
			})

})
```

Reply:
That will work for the serice call but not the other part -- How about the TeriminateParams and CancelParams? Maybe combine one? Right now they only have one : terminateChildren and cancelChildren. We can combine them into one but could be a little confusing as the WF parameter. If we called it childPolicy(true/false/nil), it is incorrect because it doen't apply to reset/signal.

Reply:
You can just pass in a boolean to the `processChildRecursiveTask` - `applyOnChildren`.. That isnt something that depends on whether it is cancel or terminate.

Reply:
sounds good